src/ophyd_async/core/soft_signal_backend.py

```diff
@@ -74,37 +83,45 @@ def make_initial_value(self, datatype: Optional[Type[T]]) -> T:
         return cast(T, datatype(shape=0))  # type: ignore
 
 
-@dataclass
 class SoftEnumConverter(SoftConverter):
-    enum_class: Type[Enum]
+    choices: Tuple[str, ...]
 
-    def write_value(self, value: Union[Enum, str]) -> Enum:
+    def __init__(self, datatype: Union[SubsetEnum, Enum]):  # type: ignore
```

Comment:
...then this could be:
```suggestion
    def __init__(self, datatype: Union[RuntimeSubsetEnum, Enum]):
```
and `datatype.choices` will be type checked